### PR TITLE
[NPM-3444] Include conntrack dump output in agent flares

### PR DESCRIPTION
**Pull request by pimlu**
### What does this PR do?

This PR adds two files to the agent flare:

```
system_probe/conntrack_cached.log: queries /network_tracer/debug/conntrack/cached
system_probe/conntrack_host.log: queries /network_tracer/debug/conntrack/host
```

The format looks like this:
```
conntrack dump, kind=cached-nat rootNS=1234
totalEntries=2
namespace 1234, size=2:
TCP v4 src=10.0.0.1 dst=10.0.0.2 sport=11000 dport=11001 src=10.0.0.3 dst=10.0.0.4 sport=11002 dport=11003
TCP v4 src=127.0.0.1 dst=127.0.0.2 sport=10000 dport=10001 src=127.0.0.3 dst=127.0.0.4 sport=10002 dport=10003
```

The table is limited to 3000 entries.

### Motivation

There have been a number of support cases where having the conntrack table would have helped.

### Describe how to test/QA your changes

You can watch the result of a conntrack dump by doing:
```
# create a no-op rule to force conntrack to start tracking connections
sudo iptables -A INPUT -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
# watch the conntrack output
sudo watch curl --unix-socket /var/run/sysprobe.sock http://unix/network_tracer/debug/conntrack/host
```

The same command works for `conntrack/cached` (which is the same as `/conntrack/host` but usually implemented using eBPF).

To create an artificial NAT connection, set up iptables with the following:

```
link="dummy-$RANDOM"
sudo ip link add $link type dummy
sudo ip address add 1.1.1.1 broadcast + dev $link
sudo ip link set $link up
sudo iptables -t nat -A OUTPUT --dest 2.2.2.2 -j DNAT --to-destination 1.1.1.1
```
Then run `python3 -m http.server` which will listen on the dummy interface, and `curl 2.2.2.2:8000` will create a NAT'd connection.

The agent flare is also expected to contain these results, listed in the "What does this PR do?" section.

### CODE REVIEW

**Comment by pimlu** [comment]
Publishing for review, although I am still working on how to test NAT'd connections

File: releasenotes/notes/conntrack-in-flare-859fcd25497e4b6c.yaml
Comment (buraizu) [inline releasenotes/notes/conntrack-in-flare-859fcd25497e4b6c.yaml:11]:
```suggestion
    The Agent flare now returns NAT debug information from conntrack in the ``system-probe`` directory.
```

**Review by buraizu — APPROVED** [review]
Approving with a minor update requested to the release note

File: pkg/network/tracer/tracer_dump_conntrack.go
Comment (hmahmood) [inline pkg/network/tracer/tracer_dump_conntrack.go:44]: Is there a way we can enforce this earlier? Like for the netlink based dump, stop reading after maxEntries?

File: pkg/flare/sysprobe/archive_sysprobe_linux.go
Comment (pimlu) [reply pkg/flare/sysprobe/archive_sysprobe_linux.go:32]: Okay, will remove

File: pkg/network/tracer/tracer_dump_conntrack.go
Comment (pimlu) [reply pkg/network/tracer/tracer_dump_conntrack.go:44]: I think we potentially could by closing the netlink socket prematurely. My original intention with this was to make the endpoint deterministic by pulling the full table and then sorting, so that you can visually see the difference (or run `git diff conntrack_host.log conntrack_cached.log`). But if that's not useful, I can remove the sorting/close it early.

File: pkg/network/tracer/tracer_dump_conntrack.go
Comment (pimlu) [reply pkg/network/tracer/tracer_dump_conntrack.go:62]: Responded in [this comment](https://github.com/DataDog/datadog-agent/pull/30157#discussion_r1803834225)

File: pkg/flare/sysprobe/archive_sysprobe_linux.go
Comment (ogaca-dd) [inline pkg/flare/sysprobe/archive_sysprobe_linux.go:6]: Can you update `.github/CODEOWNERS` to set the proper owner for `pkg/sysprobe`.

File: pkg/network/tracer/tracer_dump_conntrack.go
Comment (pimlu) [reply pkg/network/tracer/tracer_dump_conntrack.go:50]: Updated to close the netlink socket after 10s and return partial results. Relevant bits:

* Updated with [10 second timeout](https://github.com/DataDog/datadog-agent/pull/30157/files#diff-18d9eb49206512ac164798eb2f808d958d5bc6acbb3a8e50b447bd51f2b4deccR270)
* Now [returns partial results](https://github.com/DataDog/datadog-agent/pull/30157/files#diff-e4de21a6ad4c71906d7fe6406df6001147c0600e791529170d71fe32bfa1bafeR130) after a timeout
* Removed the host_full stuff that dumped the whole conntrack table
* [Added a check](https://github.com/DataDog/datadog-agent/pull/30157/files#diff-d55acaf7f363a35345fe9cceca15bb12a45cbf1b31d05a91853f4d285fb6361aR812) for when a timeout occurred and [include that info in the dump](https://github.com/DataDog/datadog-agent/pull/30157/files#diff-ee208cb7d4a1e151f8e433fe4a893db9603713162cd10ead9eb8e4b0b671665fR61)

**Review by hmahmood — APPROVED** [review]

**Review by akarpz — APPROVED** [review]

File: pkg/flare/sysprobe/archive_linux.go
Comment (guyarb) [inline pkg/flare/sysprobe/archive_linux.go:4]:
```suggestion
// Copyright 2024-present Datadog, Inc.
```
please change accordingly in all new files

File: .github/CODEOWNERS
Comment (guyarb) [inline .github/CODEOWNERS:448]: suggestion - as sysprobe is very generic, let's add all system-probe teams as owners to the directory
or let's rename this package to indicate it belongs only with NPM

File: pkg/flare/sysprobe/archive_linux.go
Comment (pimlu) [reply pkg/flare/sysprobe/archive_linux.go]: Updated, thanks

File: .github/CODEOWNERS
Comment (pimlu) [reply .github/CODEOWNERS]: Thanks, added other system-probe teams as owners - did I miss anybody?

**Review by derekwbrown — APPROVED** [review]
approved for one windows file

**Review by iglendd — APPROVED** [review]

**Review by robertjli — APPROVED** [review]

**Review by pgimalac — APPROVED** [review]

**Pull request merged**
